flang/lib/Semantics/resolve-directives.cpp

```diff
@@ -1910,7 +1910,8 @@ void OmpAttributeVisitor::Post(const parser::Name &name) {
       if (Symbol * found{currScope().FindSymbol(name.source)}) {
         if (symbol != found) {
           name.symbol = found; // adjust the symbol within region
-        } else if (GetContext().defaultDSA == Symbol::Flag::OmpNone) {
+        } else if (GetContext().defaultDSA == Symbol::Flag::OmpNone &&
+            !symbol->test(Symbol::Flag::OmpThreadprivate)) {
```

Comment:
Would including Threadprivate also into `IsObjectWithDSA` work?

Reply:
Hi Kiran, Apologies for the late reply, I was on vacation. 

The symbol is already being added to objectWithDSA map in OmpAttributeVisitor::Pre(const parser::OpenMPThreadprivate)->ResolveOmpObjectList()->ResolveOmpObject()->AddToContextObjectWithDSA(), yet in OmpAttributeVisitor::Post(const parser::Name) the same symbol when checked in IsObjectWithDSA() is getting false.